kajas/src/app/forgot-password/forgot-password.component.ts

```diff
@@ -0,0 +1,129 @@
+import { Component, OnInit, OnDestroy, Renderer2 } from '@angular/core';
+import { FormBuilder, FormGroup, Validators, AbstractControl, ValidationErrors } from '@angular/forms';
+import { Router } from '@angular/router';
+import { ModalService } from '../../services/modal.service'; 
+import axios from 'axios';
+
+@Component({
+  selector: 'app-forgot-password',
+  templateUrl: './forgot-password.component.html',
+  styleUrls: ['./forgot-password.component.css']
+})
+
+export class ForgotPasswordComponent implements OnInit, OnDestroy {
+  forgotPasswordForm: FormGroup;
+  submitted = false;
+  errorMessage: string = '';
+  showModal: boolean = false;
+  modalMessage: string = '';
+
+  constructor(
+    private renderer: Renderer2,
+    private fb: FormBuilder,
+    private router: Router,
+    private modalService: ModalService 
+  ) {
+    this.forgotPasswordForm = this.fb.group({
+      email: ['', [Validators.required, Validators.email, this.gmailValidator]]
+    });
+  }
+
+  ngOnInit(): void {
+    this.setInitialStyles();
+    window.addEventListener('resize', this.applyBackground.bind(this));
+  }
+
+  ngOnDestroy(): void {
+    this.revertStyles();
+    window.removeEventListener('resize', this.applyBackground.bind(this));
```

Comment:
### Stop using Vanilla JS way of doing things.
- Get rid of using querySelectors and addEventsListener.
[https://docs.google.com/presentation/d/1Mdcq3WcHksBlbCvxi6wx5vajn4v8LGlJRnBIu3x4fvw/edit#slide=id.g2e205e48566_0_29](url)